kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/records/BatchAwareMemoryRecordsBuilder.java

```diff
@@ -144,6 +145,21 @@ public BatchAwareMemoryRecordsBuilder addBatch(CompressionType compressionType,
                 templateBatch.deleteHorizonMs().orElse(RecordBatch.NO_TIMESTAMP));
     }
 
+    /**
+     * Directly appends a batch, intended to be used for passing through unmodified batches. Writes
+     * the previous batch to the stream if required
+     * @param templateBatch The batch to use as a source of batch parameters
+     * @return this builder
+     */
+    public @NonNull BatchAwareMemoryRecordsBuilder writeBatch(MutableRecordBatch templateBatch) {
+        if (haveBatch()) {
+            appendCurrentBatch();
+        }
+        templateBatch.writeTo(buffer);
+        builder = null;
```

Comment:
should add a comment here, I did this to ensure the buffer was flipped at the end if this is the last batch added.